librubyfmt/src/render_targets.rs

```diff
@@ -3,11 +3,58 @@ use crate::line_tokens::LineToken;
 use crate::types::{ColNumber, LineNumber};
 use std::collections::HashSet;
 
+fn insert_at<T>(idx: usize, target: &mut Vec<T>, input: &mut Vec<T>) {
+    let drain = input.drain(..);
+    let mut idx = idx;
+    for item in drain {
+        target.insert(idx, item);
+        idx += 1;
+    }
+}
+
+#[derive(Copy, Clone, Debug)]
 pub enum ConvertType {
     MultiLine,
     SingleLine,
 }
 
+pub trait LineTokenTarget {
+    fn push(&mut self, lt: LineToken);
+    fn insert_at(&mut self, idx: usize, tokens: &mut Vec<LineToken>);
+    fn into_tokens(self, ct: ConvertType) -> Vec<LineToken>;
+    fn last_token_is_a_newline(&self) -> bool;
+    fn index_of_prev_hard_newline(&self) -> Option<usize>;
+}
```

Comment:
The impl of every method on this trait is exactly the same (or could be). How about this instead?
```suggestion
pub trait LineTokenTarget {
    fn tokens(&self) -> &[LineToken];
    fn tokens_mut(&mut self) -> &mut Vec<LineToken>;
    
    fn push(&mut self, lt: LineToken) {
        self.tokens_mut().push(lt);
    }
    
    // etc
}
```